### PR TITLE
fix: Ensure interactive embedding CTA matches instance features

### PR DESCRIPTION
**Pull request by oisincoveney**
@metabase-bot backport release-x.49.x

### Description

Ensures that the interactive embedding CTA displays the correct banner for OSS, starter, and pro plans.

### How to verify

On OSS and Starter plans, users should see:
image

On pro plans, users should see:
image

### Checklist

- [X] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by WiNloSt** [comment]
@metabase-bot backport release-x.49.x

**Review by heypoom — APPROVED** [review]
Logic looks correct to me. Thanks!

**Review by albertoperdomo — COMMENTED** [review]
Left a comment about the plan logic

File: enterprise/frontend/src/metabase-enterprise/embedding/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { getPlan } from "metabase/common/utils/plan";
+import { getSetting } from "metabase/selectors/settings";
+import type { EnterpriseState } from "metabase-enterprise/settings/types";
+
+export const isInteractiveEmbeddingEnabled = (state: EnterpriseState) => {
+  const plan = getPlan(getSetting(state, "token-features"));
+  return plan === "pro-cloud" || plan === "pro-self-hosted";
```
Comment (albertoperdomo) [inline enterprise/frontend/src/metabase-enterprise/embedding/selectors.ts:7]: Does this cover potentially other plans that have access to this feature? I am thinking (1) Enterprise but also (2) any other legacy plans that should have access to this feature?

I can ask about (2).

**Review by albertoperdomo — APPROVED** [review]

**Pull request merged**